edge-client/subdevice_fota.cpp

```diff
@@ -0,0 +1,292 @@
+/*
+ * ----------------------------------------------------------------------------
+ * Copyright 2021 Pelion Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ----------------------------------------------------------------------------
+ */
+
+#ifdef MBED_EDGE_SUBDEVICE_FOTA
+
+#include "edge-client/subdevice_fota.h"
+
+static fota_context_t* fota_ctx = NULL;
+static char *endpoint = NULL;
+
+int subdevice_init_buff() {
+    fota_ctx = (fota_context_t*) calloc(1, sizeof(fota_context_t));
+    if(fota_ctx == NULL) {
+        FOTA_TRACE_ERROR("Unable to allocate FOTA ctx.");
+        return FOTA_STATUS_OUT_OF_MEMORY;
+
+    }
+    fota_ctx->fw_info = (manifest_firmware_info_t*) calloc(1, sizeof(manifest_firmware_info_t));
+    if (!fota_ctx->fw_info) {
+        FOTA_TRACE_ERROR("Unable to allocate FW info.");
+        return FOTA_STATUS_OUT_OF_MEMORY;
+    }
+    endpoint = (char*) calloc(ENDPOINT_SIZE,1*sizeof(char));
+    if(!endpoint) {
+        FOTA_TRACE_ERROR("Unable to allocate memory for endpoint");
+        return FOTA_STATUS_OUT_OF_MEMORY;      
+    }
+    return 0;
+}
+
+int get_component_name(char* c_name) {
+    if(fota_ctx) {
+        memcpy(c_name, fota_ctx->fw_info->component_name, FOTA_COMPONENT_MAX_NAME_SIZE);
+        return 0;
+    }
+    else
+        return 0;
```

Comment:
@prakharbhatt007 Isn't the `else` redundant? 
```suggestion
    }
    return 0;
```